source/Octopus.Server.Client/IOctopusAsyncClient.cs

```diff
@@ -86,8 +85,50 @@ public interface IOctopusAsyncClient : IDisposable
         /// <param name="path">The path from which to fetch the resources.</param>
         /// <param name="pathParameters">If the <c>path</c> is a URI template, parameters to use for substitution.</param>
         /// <returns>The collection of resources from the server.</returns>
+        [Obsolete("Please use the overload with cancellation token instead.", false)]
         Task<ResourceCollection<TResource>> List<TResource>(string path, object pathParameters = null);
 
+        /// <summary>
+        /// Fetches a collection of resources from the server using the HTTP GET verb. The collection itself will usually be
+        /// limited in size (pagination) and links to the next page of data is available in the <see cref="Resource.Links" />
+        /// property.
+        /// </summary>
+        /// <exception cref="OctopusSecurityException">
+        /// HTTP 401 or 403: Thrown when the current user's API key was not valid, their
+        /// account is disabled, or they don't have permission to perform the specified action.
+        /// </exception>
+        /// <exception cref="OctopusServerException">
+        /// If any other error is successfully returned from the server (e.g., a 500
+        /// server error).
+        /// </exception>
+        /// <exception cref="OctopusValidationException">HTTP 400: If there was a problem with the request provided by the user.</exception>
+        /// <exception cref="OctopusResourceNotFoundException">HTTP 404: If the specified resource does not exist on the server.</exception>
+        /// <param name="path">The path from which to fetch the resources.</param>
+        /// <param name="cancellationToken">The request cancellation token.</param>
+        /// <returns>The collection of resources from the server.</returns>
+        Task<ResourceCollection<TResource>> List<TResource>(string path, CancellationToken cancellationToken);
```

Comment:
Where possible, the goal is to maintain similar optional parameter support as per original contract.